osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs

```diff
@@ -263,6 +257,12 @@ private int calculateEffectiveMissCount()
         }
 
         private double getComboScalingFactor() => Attributes.MaxCombo <= 0 ? 1.0 : Math.Min(Math.Pow(scoreMaxCombo, 0.8) / Math.Pow(Attributes.MaxCombo, 0.8), 1.0);
+
+        private double calculateMissPenalty(double missCount, double strainCount)
+        {
+            return 0.95 / ((missCount / (3 * Math.Sqrt(strainCount))) + 1);
```

Comment:
This probably need some comments

Reply:
not really sure what I could comment here, and I don't think it'd end up being very helpful anyway? have you got anything specific in mind

Reply:
Something about why exactly you need difficult strains here would be good I think. Basically just explain the idea so that people that have no idea what's going on could understand the motivation for this penalty